brave/brave-core
# Fix DevToolsWindow using original profile for Tor window

Resolves https://github.com/brave/brave-browser/issues/23579

Tor profile only fulfills `IsOffTheRecord()` so we need to make it a primary OTR so it would behave same as private window.
This also disable Ambient auth for Tor.  (It is disabled in private window but not for non primary OTR profile)
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/net/profile_network_context_service.cc;l=168-172;drc=169c6cc102b39295a5bfe2f2a176b42b1c2fe2c4

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Open Tor tab. open devtools, enter a=123
2. Close Tor tab, open normal tab. open devtools, hit the up arrow. you won't be able to see a=123
